packages/collector/test/test_util/ProcessControls.js

```diff
@@ -248,33 +252,42 @@ class ProcessControls {
             }
           })
           .join('&');
-        opts.url = opts.url.includes('?') ? `${opts.url}&${queryParams}` : `${opts.url}?${queryParams}`;
+
+        requestOpts.url = requestOpts.url.includes('?')
+          ? `${requestOpts.url}&${queryParams}`
+          : `${requestOpts.url}?${queryParams}`;
       }
-      opts.json = true;
-      opts.ca = cert;
+
+      requestOpts.json = true;
+      requestOpts.ca = cert;
+
       let response;
-      try {
-        const result = await fetch(opts.url, opts);
-        const contentType = result.headers.get('content-type');
-        if (contentType && contentType.includes('application/json')) {
-          response = await result.json();
-        } else if (contentType && (contentType.includes('text/html') || contentType.includes('text/plain'))) {
-          response = await result.text();
-        } else {
-          response = {
-            body: await result.text(),
-            status: result.status,
-            headers: result.headers
-          };
-        }
 
-        return response;
```

Comment:
The function simply returned "Internal Server Error" if the status code of the app was 500.